addons/web/static/src/core/debug/profiling/profiling_qweb.js

```diff
@@ -0,0 +1,325 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+import { Field } from '@web/views/fields/field';
+import { loadBundle } from "@web/core/assets";
+import { renderToString } from "@web/core/utils/render";
+
+const { useState, useRef, onWillStart, onMounted, onWillUnmount } = owl;
+
+
+/**
+ * This widget is intended to be used on Text fields. It will provide Ace Editor
+ * for display XML and Python profiling.
+ */
+export class ProfilingQwebView extends Field {
+    setup() {
+        super.setup();
+
+        this.orm = useService("orm");
+        this.ace = useRef("ace");
+        this.selector = useRef("selector");
+        this.mode = "readonly";
+
+        // {archs, data:{template, xpath, directive, time, duration, query }[]}
+        const results = (this.props.value && JSON.parse(this.props.value)[0].results) || {
```

Comment:
results should be a getter. This way, if props were to change, results would be up to date. 
This is the simple way, there is also onWillUpdateProps and useEffect for more sophisticated situations